Upgrade to latest RN version

ajwhite opened the pull request:
Upgrades to the latest version of RN.

- Removes `Activity` injector in the constructor, now uses `getCurrentActivity` (new to `ReactContextBaseJavaModule`)
- Implements `ActivityEventListener` to receive activity results

christophermark commented on android/build.gradle:
```diff
@@ -30,5 +30,5 @@ repositories {
 
 dependencies {
-    compile 'com.facebook.react:react-native:0.15.+'
+    compile 'com.facebook.react:react-native:+'
```
We might need a minimum dependency here as it seems there were breaking changes at some point.

ajwhite replied:
How is that written for gradle? I believe the min should now be `0.20`

brianmuse replied:
I dont think there's much you can do about this at the moment other than lock into the major version.

ajwhite replied:
Damn.. That makes me really cautious about checking the versions of all the dependencies we currently use. 

how does grade work with shared libraries? If this requires, say 0.17 like it did previously, but the main app requires 0.26, do they each get their own copy, yet access the same instances during runtime?

We hit a problem where certain methods weren't accessible by this package because of the versioning differences. So it seemed like they each got their own specific RN lib as a dep

christophermark commented on android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java:
```diff
@@ -1,28 +1,30 @@
 import com.facebook.react.bridge.WritableNativeMap;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
 
+
```
Intentional newline add?

ajwhite replied:
Noop

christophermark commented:
A few small things, LGTM otherwise!

brianmuse commented on android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java:
```diff
@@ -84,20 +90,20 @@ public void onPermissionResult() {
+                    Uri.parse("package:" + application.getPackageName())
             );
-            activity.startActivityForResult(intent, writeSettingsRequestCode);
+            application.startActivityForResult(intent, writeSettingsRequestCode, null);
```
You might be able to simplify some of this with `ActivityCompat.requestPermissions`

https://github.com/robinpowered/robin-compass/blob/develop/android/app/src/main/java/com/robinpowered/compass/PermissionRequestActivity.java#L31

ajwhite replied:
This doesn't actually prompt a permission dialog - only redirects to a settings page. This is a permission that can't be done by a dialog apparently. Does that change anything regarding your feedback?

brianmuse replied:
Oh interesting. Yeah fine as-is then.

brianmuse commented on android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java:
```diff
@@ -84,20 +90,20 @@ public void onPermissionResult() {
         return Build.VERSION.SDK_INT < Build.VERSION_CODES.M ||
                 (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
-                Settings.System.canWrite(getReactApplicationContext()));
+                        Settings.System.canWrite(getReactApplicationContext()));
```
You can likely simplify this with `ContextCompat`

https://github.com/robinpowered/robin-compass/blob/779ed696f8d09cbf586a16119011a257dc883fd3/android/app/src/main/java/com/robinpowered/compass/reactnative/PermissionManager.java#L168

brianmuse replied:
Does this also not work?

ajwhite replied:
I can give it a try with your suggestion, maybe it works for checking

brianmuse replied:
Update on this? would like to get merged for you

brianmuse commented on android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java:
```diff
@@ -194,8 +200,13 @@ public void getSystemBrightness(final Promise promise) {
-        float brightness = activity.getWindow().getAttributes().screenBrightness;
-        promise.resolve(brightness);
+        Activity activity = getCurrentActivity();
+        if (activity != null) {
```
Is this possible here?

ajwhite replied:
It's a nullable, so I think it should be checked. They specifically flagged it as such, and it's only a getter that subclasses (native modules) call, so I think we are supposed to check.

brianmuse replied:
👍

ajwhite commented:
Updated to use `ContetxCompat` for permission checking.

Also updated the gradle RN version to be `react-native:+`, since the the version is actually installed by `npm` via the `package.json` version, `0.27`. Gradle just pulls that in, I believe, so we should be safe there.

christophermark commented:
👍

brianmuse commented:
Aight yeah lets go! 👍
